src/Carnac.Logic/ShortcutCollection.cs

```diff
@@ -8,12 +8,12 @@ namespace Carnac.Logic
     public class ShortcutCollection : Collection<KeyShortcut>
     {
         public string Group { get; set; }
+
         public string Process { get; set; }
 
-        public IEnumerable<KeyShortcut> GetShortcutsMatching(IEnumerable<KeyPress> keys)
+        public KeyShortcut[] GetShortcutsMatching(IEnumerable<KeyPress> keys)
         {
-            var matches = this.Where(s => s.StartsWith(keys));
-            return matches;
+            return this.Where(s => s.StartsWith(keys)).ToArray();
```

Comment:
Arrays are still mutable, but using a `List` here is slightly smarter in terms of memory allocation (because it doesn't attempt to trim the backing store to the length of the array).